src/diffusers/loaders/lora_pipeline.py

```diff
@@ -1968,7 +1972,25 @@ def _maybe_expand_transformer_param_shape_or_error_(
         is_peft_loaded = getattr(transformer, "peft_config", None) is not None
         for name, module in transformer.named_modules():
             if isinstance(module, torch.nn.Linear):
-                module_weight = module.weight.data
+                is_bnb_4bit_quantized = module.weight.__class__.__name__ == "Params4bit"
+                if is_bnb_4bit_quantized and not is_bitsandbytes_available():
+                    raise ValueError(
+                        "The checkpoint seems to have been quantized with `bitsandbytes` (4bits). Install `bitsandbytes` to load quantized checkpoints."
+                    )
+                elif is_bnb_4bit_quantized:
+                    weight_on_cpu = False
+                    if not module.weight.is_cuda:
+                        weight_on_cpu = True
+                    module_weight = dequantize_bnb_weight(
+                        module.weight.cuda() if weight_on_cpu else module.weight,
+                        state=module.weight.quant_state,
+                        dtype=transformer.dtype,
+                    ).data
+                    if weight_on_cpu:
+                        module_weight = module_weight.cpu()
+                    module_weight = dequantize_bnb_weight(module.weight, state=module.weight.quant_state).data
```

Comment:
Just a question. Why dequantize again after moving the weight to CPU? Aren't we overwriting the previous step?